tests/test_expression.h

```diff
@@ -0,0 +1,419 @@
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_EXPRESSION_H
+#define TEST_EXPRESSION_H
+
+#include "core/math/expression.h"
+
+#include "thirdparty/doctest/doctest.h"
+
+namespace TestExpression {
+
+TEST_CASE("[Expression] Integer arithmetic") {
+	Expression expression;
+
+	CHECK_MESSAGE(
+			expression.parse("-123456") == OK,
+			"Integer identity should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == -123456,
+			"Integer identity should return the expected result.");
+
+	CHECK_MESSAGE(
+			expression.parse("2 + 3") == OK,
+			"Integer addition should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == 5,
+			"Integer addition should return the expected result.");
+
+	// FIXME: This doesn't behave as expected due to integer overflow.
+	//
+	//CHECK_MESSAGE(
+	//		expression.parse("999999999999 + 999999999999") == OK,
+	//		"Large integer addition should parse successfully.");
+	//CHECK_MESSAGE(
+	//		int(expression.execute()) == 1'999'999'999'998,
```

Comment:
This expression is working correctly for me, and returns `1999999999998`, but you are using `int` (32-bit on most systems) operator instead of `int64_t`.

```suggestion
  //		int64_t(expression.execute()) == 1'999'999'999'998,
```